File: longest-substring-without-repeating-characters.cpp
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> st = new HashSet<Character>();
        int i=0,j=0,maxlen=0;
        while(i<s.length() && j<s.length()){
            if(!st.contains(s.charAt(j))){
                st.add(s.charAt(j));
                j++;
                maxlen=Math.max(maxlen,j-i);
            }
            else{
                st.remove(s.charAt(i));
                i++;
            }
        }
        return maxlen;
    }
}

File: longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int maxi=0;
        int l=0,r=0;
        HashSet<Character> st = new HashSet<>();
        while(r<n){
            if(!st.contains(s.charAt(r))){
                st.add(s.charAt(r));
                maxi=Math.max(maxi,r-l+1);
                r++;
            }
            else{
                st.remove(s.charAt(l));
                l++;
            }
        }
        return maxi;
    }
}